### ISSUE TITLE
Consistent approach to develop/master branch

### ISSUE THREAD

**Issue by mattheworiordan** [ably/docs#739]
This repo does not follow the development flow, see https://wiki.ably.io/pages/Development-flow-for-Ably-repos.html.

If we did have a `develop` branch though, would that get published to `docs.ably.io` or would we have `docs-dev.ably.io` and `docs.ably.io` which is `master`.

┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/DOC-126) by [Unito](https://www.unito.io)

**Comment by paddybyers** [comment]
> would we have `docs-dev.ably.io` and `docs.ably.io` which is `master`.

I do think there is an explicit step between

1) merge a PR to develop, signifying approval for it to be incorporated into future specs;
2) show that change on the docs website. Perhaps that would need also to be gated on getting those spec changes into the spreadsheet.

So doing a ff on `master` achieves (2), if `docs.ably.io` points to `master`.

Not sure if we need `docs-dev`. Wouldn't we be asking client library devs to make changes once it's on master?

**Comment by mattheworiordan** [comment]
> merge a PR to develop, signifying approval for it to be incorporated into future specs;

Ok, although isn't it signifying either a) it can be incorporated into current releases, b) it is gated to a release i.e. needs to be merged into a particular branch.

> show that change on the docs website. Perhaps that would need also to be gated on getting those spec changes into the spreadsheet.

Please see https://github.com/ably/docs/issues/566#issuecomment-460963437

> So doing a ff on master achieves (2), if docs.ably.io points to master.

Well that happens now anyway.

> Not sure if we need docs-dev. Wouldn't we be asking client library devs to make changes once it's on master?

Sure, but we tend to do this for all other repos i.e. have `[staging|sandbox|dev]-[domain]` track `develop` branch, and with Heroku is trivial to do.  So perhaps worth doing for consistency?
